src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs

```diff
@@ -81,7 +81,14 @@ public static IResourceBuilder<ContainerResource> AddContainer(this IDistributed
         return builder.WithAnnotation(annotation);
     }
 
-    public static IResourceBuilder<T> WithArgs<T>(this IResourceBuilder<T> builder, params string[] args) where T : IResource
+    /// <summary>
+    /// Adds the arguments to be passed to a container resource when the container is started.
+    /// </summary>
+    /// <typeparam name="T">The resource type.</typeparam>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="args">The arguments to be passed to the container when it is started.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>    
+    public static IResourceBuilder<T> WithArgs<T>(this IResourceBuilder<T> builder, params string[] args) where T : ContainerResource
```

Comment:
I'm assuming the change in the generic constraint is intentional? I ask since PR really only mentions xml docs

Reply:
I don't think this is meant to be constrained. @davidfowl @mitchdenny